src/main/java/com/seatapp/services/SeatService.java

```diff
@@ -62,17 +69,23 @@ public List<Seat> getAll() {
 
     /**
      * Reserves the seat with the specified id.
-     * @param seatId is the id of the to be reserved seat.
+     *
+     * @param seatId         is the id of the to be reserved seat.
+     * @param reservationDto is the reservation details.
      * @return the reserved seat.
      */
-    public Seat reserve(final Long seatId) {
+    public Seat reserve(final Long seatId,
+                        final ReservationDto reservationDto) {
+        if (reservationDto == null) {
+            throw new IllegalArgumentException("ReservationDto cannot be null");
+        }
         Seat seat = seatRepository.findById(seatId)
```

Comment:
This is used multiple times in different methods.
Create a seperate method for this.